packages/kbn-esql-utils/src/utils/run_query.ts

```diff
@@ -61,32 +62,22 @@ export async function getESQLQueryColumns({
   }
 }
 
-export async function getESQLQueryColumnsRaw({
+// Returns the columns with the kibana format
+// Based on the search api from the data plugin
+export async function getESQLQueryColumns({
   esqlQuery,
   search,
   signal,
 }: {
   esqlQuery: string;
   search: ISearchGeneric;
   signal?: AbortSignal;
-}): Promise<ESQLColumn[]> {
+}): Promise<DatatableColumn[]> {
   try {
-    const response = await lastValueFrom(
-      search(
-        {
-          params: {
-            query: `${esqlQuery} | limit 0`,
-            version: ESQL_LATEST_VERSION,
-          },
-        },
-        {
-          abortSignal: signal,
-          strategy: 'esql_async',
-        }
-      )
-    );
+    const rawColumns = await getESQLQueryColumnsRaw({ esqlQuery, search, signal });
```

Comment:
```suggestion
    const rawColumns = await getESQLQueryColumnsRaw({ esqlQuery: `${esqlQuery} | limit 0`, search, signal });
```

Should we keep `limit 0`?

Reply:
We are doing this inside the function so no need to set it twice

```
export async function getESQLQueryColumnsRaw({
  esqlQuery,
  search,
  signal,
}: {
  esqlQuery: string;
  search: ISearchGeneric;
  signal?: AbortSignal;
}): Promise<ESQLColumn[]> {
  try {
    const response = await lastValueFrom(
      search(
        {
          params: {
            query: `${esqlQuery} | limit 0`,
            version: ESQL_LATEST_VERSION,
          },
        },
        {
          abortSignal: signal,
          strategy: 'esql_async',
        }
      )
    );

    return (response.rawResponse as unknown as ESQLSearchReponse).columns ?? [];
  } catch (error) {
    throw new Error(
      i18n.translate('esqlUtils.columnsErrorMsg', {
        defaultMessage: 'Unable to load columns. {errorMessage}',
        values: { errorMessage: error.message },
      })
    );
  }
}
```